ui/app/mainui/adaptors/AllContactsAdaptor.qml

```diff
@@ -0,0 +1,97 @@
+            "lastUpdated", "lastUpdatedLocally", "thumbnailImage", "largeImage",
+            "isContactRequestReceived", "isContactRequestSent", "removed",
+            "trustStatus", "bio"
+        ]
+
+        markerRoleName: ""
+
+        sources: [
+            SourceModel {
+                model: ObjectProxyModel {
+                    sourceModel: ListModel {
+                        ListElement {
+                            _: "" // empty role to prevent warning
+                        }
+                    }
+
+                    delegate: QtObject {
+                        readonly property string pubKey: root.selfPubKey
+                        readonly property string displayName: root.selfDisplayName
+                        readonly property string ensName: root.selfName
+                        readonly property bool isEnsVerified: root.selfName !== ""
+                                                              && Utils.isValidEns(root.selfName)
+                        readonly property string localNickname: ""
+                        readonly property string preferredDisplayName: root.selfPreferredDisplayName
+                        readonly property string name: preferredDisplayName
+                        readonly property string alias: root.selfAlias
+                        readonly property string icon: root.selfIcon
+                        readonly property int colorId: root.selfColorId
+                        readonly property var colorHash: root.selfColorHash
+                        readonly property int onlineStatus: root.selfOnlineStatus
+                        readonly property bool isContact: false
+                        readonly property bool isCurrentUser: true
+                        readonly property bool isVerified: false
+                        readonly property bool isUntrustworthy: false
+                        readonly property bool isBlocked: false
+                        readonly property int contactRequestState: Constants.ContactRequestState.None
+                        readonly property int lastUpdated: 0
+                        readonly property int lastUpdatedLocally: 0
+                        readonly property string thumbnailImage: root.selfThumbnailImage
+                        readonly property string largeImage: root.selfLargeImage
+                        readonly property bool isContactRequestReceived: Constants.ContactRequestState.None
+                        readonly property bool isContactRequestSent: Constants.ContactRequestState.None
+                        readonly property bool removed: false
+                        readonly property int trustStatus: Constants.trustStatus.unknown
+                        readonly property string bio: root.selfBio
+                    }
+
+                    exposedRoles: concatModel.expectedRoles
+                }
+            },
+            SourceModel {
```

Comment:
What is `SourceModel`? :thinking:

Reply:
`SourceModel` is auxiliary type for `ConcatModel` wrapping the actual concatenated model.

From `ConcatModelPage`:
```qml
    ConcatModel {
        id: concatModel

        sources: [
            SourceModel {
                model: firstModel
                markerRoleValue: "first_model"
            },
            SourceModel {
                model: secondModel
                markerRoleValue: "second_model"
            }
        ]

        markerRoleName: "which_model"
        expectedRoles: ["key", "name"]
    }
```